helper/helper.go

```diff
@@ -0,0 +1,83 @@
+package helper
+
+import (
+	"archive/tar"
+	"errors"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+func GetApplication(applicationsPath, applicationUUID string) (application, commit string, err error) {
+	commit, err = getCommit(applicationsPath, applicationUUID)
+	if err != nil {
+		log.Printf("Failed to get commit. Error: %s\r\n", err)
+		return "", "", err
+	}
+
+	basePath := filepath.Join(applicationsPath, applicationUUID, commit)
+	tarball := filepath.Join(basePath, "binary.tar")
+	application = filepath.Join(basePath, "application.zip")
+
+	_, err = os.Stat(application)
+	if err != nil {
+		err = untar(tarball, basePath)
+		if err != nil {
+			log.Printf("Failed to extract application. Error: %s\r\n", err)
+			return "", "", err
+		}
+	}
+
+	return
+}
+
+func untar(tarball, target string) error {
+	reader, err := os.Open(tarball)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+	tarReader := tar.NewReader(reader)
+
+	for {
+		header, err := tarReader.Next()
```

Comment:
Why is this not in the `for` loop? Very odd to have immediate conditions. This would be better as:

```go
for header, err := tarReader.Next(); err != io.EOF; header, err = tarReader.Next() {
    ...
}
```